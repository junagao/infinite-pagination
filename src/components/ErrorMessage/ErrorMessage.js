import React from 'react'
import { string } from 'prop-types'
import Text from './ErrorMessage.styles'

export default function ErrorMessage({ text }) {
  return (
    <Text data-testid="error-message">
      The following error occurred: {text}. Please try again.
    </Text>
  )
}

ErrorMessage.propTypes = {
  text: string.isRequired,
}
