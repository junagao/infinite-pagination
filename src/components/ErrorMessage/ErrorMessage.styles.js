import styled from '@emotion/styled'

const Text = styled.p`
  font-size: ${({ theme }) => theme.fontSizes.md};
  color: ${({ theme }) => theme.colors.error};
  padding: ${({ theme }) => `${theme.space[6]} ${theme.space[0]}`};
`

export default Text
