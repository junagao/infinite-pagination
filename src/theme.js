const theme = {
  colors: {
    background: '#fff',
    primary: '#111',
    text: { body: '#111', hover: '#f0f0f0' },
    disabled: '#ddd',
    error: '#BD271E',
  },
  fontSizes: {
    xl: '4rem',
    lg: '2rem',
    md: '1.5rem',
    sm: '1rem',
    xs: '0.75rem',
  },
  space: [
    0,
    '0.25rem',
    '0.5rem',
    '0.75rem',
    '1rem',
    '1.5rem',
    '2rem',
    '4rem',
    '6rem',
  ],
  borders: {
    none: 'none',
    thin: '1px solid',
  },
  radii: {
    none: 0,
    base: '0.25em',
    round: '99999em',
  },
}

export default theme
