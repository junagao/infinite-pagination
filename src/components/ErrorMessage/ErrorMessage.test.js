import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'emotion-theming'
import { matchers } from 'jest-emotion'
import theme from '../../theme'
import ErrorMessage from './ErrorMessage'

expect.extend(matchers)

describe('ErrorMessage', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <ErrorMessage text="ERROR" />
      </ThemeProvider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('displays the correct error message', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <ErrorMessage text="ERROR" />
      </ThemeProvider>,
    )
    expect(getByTestId('error-message')).toHaveTextContent(
      'The following error occurred: ERROR. Please try again.',
    )
  })

  test('renders the error message with the correct styles from theme', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <ErrorMessage text="ERROR" />
      </ThemeProvider>,
    )
    expect(getByTestId('error-message')).toHaveStyleRule('font-size', '1.5rem')
    expect(getByTestId('error-message')).toHaveStyleRule('color', '#BD271E')
    expect(getByTestId('error-message')).toHaveStyleRule('padding', '2rem 0')
  })
})
