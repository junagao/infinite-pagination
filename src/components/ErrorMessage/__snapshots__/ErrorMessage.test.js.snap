// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage matches snapshot 1`] = `
.emotion-0 {
  font-size: 1.5rem;
  color: #BD271E;
  padding: 2rem 0;
}

<p
  class="emotion-0 emotion-1"
  data-testid="error-message"
>
  The following error occurred: 
  ERROR
  . Please try again.
</p>
`;
